tools/runners/VeribleExtractor.py

```diff
@@ -0,0 +1,66 @@
+from BaseRunner import BaseRunner, kill_child_processes, set_process_limits
+
+import os
+import subprocess
+
+
+class VeribleExtractor(BaseRunner):
+    def __init__(self):
+        super().__init__(
+            "verible_extractor", "verible-verilog-kythe-extractor",
+            {"parsing"})
+
+        self.url = "https://github.com/google/verible"
+
+    def prepare_run_cb(self, tmp_dir, params):
+        src_list_path = os.path.join(tmp_dir, "src_list")
+        src_list = open(src_list_path, "w")
+        print("\n".join(params.get("files", [])), file=src_list)
+
+        inc_dirs = ",".join(params.get("incdirs", []))
+
+        self.cmd = [
+            self.executable, '--file_list_root', '', '--include_dir_paths',
+            inc_dirs, '--file_list_path', src_list_path
+        ]
+
+    def run_subprocess(self, tmp_dir, params):
+        self.prepare_run_cb(tmp_dir, params)
+
+        proc = subprocess.Popen(
+            self.cmd,
+            cwd=tmp_dir,
+            preexec_fn=set_process_limits,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.PIPE)
+
+        timeout = int(params['timeout'])
+        if 'DISABLE_TEST_TIMEOUTS' in os.environ:
+            timeout = None
+        else:
+            try:
+                timeout = int(os.environ['OVERRIDE_TEST_TIMEOUTS'])
+            except KeyError:
+                # continue with timeout from params
+                pass
+            except ValueError:
+                return ("Invalid OVERRIDE_TEST_TIMEOUTS value", 1)
+
+        try:
+            _, log = proc.communicate(timeout=timeout)
+            # verible-verilog-kythe-extractor always exits with rc=0.
+            # Check error log instead - it should be empty when everything
+            # succeeded.
+            returncode = 0 if not log else 1
```

Comment:
Is this the only reason we have to copy the whole `run_subprocess` method?
Why do we completely remove stdout in this runner? Is there a way we can preserve it?
Maybe if you do have to analyze the output you could do sth like
```
def run_subprocess(self, tmp_dir, params):
    super.run_subprocess(self, tmp_dir, params)
    [ analyze the output here and change the return code ]
```

The thing I want to avoid is having to maintain a very similar code in two places

Reply:
`verible-verilog-kythe-extractor` outputs JSON stream on stdout and log messages on stderr. Do we need to store the stream?
`BaseRunner.run_subprocess` assumes log is on stdout and drops stderr, so there's nothing to check.

To avoid code duplication I can modify `BaseRunner.run_subprocess` and move the code for handling process output and status checking (those two lines: https://github.com/SymbiFlow/sv-tests/blob/master/tools/BaseRunner.py#L124) to separate method.

Reply:
Ok, I understand the issue now.

Please take a look at the Verilator runner:
https://github.com/SymbiFlow/sv-tests/blob/master/tools/runners/Verilator.py#L67

It generates a helper bash script and then just runs it - you could do something similar.

You could:
* redirect stdout to /dev/null if its not needed
* analyze stderr
* exit the script with the appropriate code

You then will not have to modify `run_subprocess` at all.

Would this work?

Reply:
Done